cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterCa.java

```diff
@@ -241,4 +241,15 @@ public Secret cruiseControlSecret() {
             isMaintenanceTimeWindowsSatisfied);
     }
 
+    @Override
+    protected boolean isCaCertThumbprintChanged() {
```

Comment:
This seems weird. The method has a generic name, but it really checks only the ZooKeeper secret. Is that intentional? What about the other secrets? Also, this is a bit of a liability once we get rid of ZooKeeper.

Reply:
I think it is not just the Zoo removal what would be the problem here. One thing to keep in mind is restartability of the process. I.e., what happens if the operator fails in the middle before updating and rolling all components.

Reply:
Good point. It means that currently if after the first ZK rolls, the new CA certificate is trusted by ZK pods and the ZK Secret is updated with the new thumbprint but the CO crashes, it won't recover to roll K and the other pods. The simpler solution I see is about checking the thumbprint of CA certificate on all the Secrets in the `ClusterCA` to be sure that if at least one is different, the `renewalType=REPLACE_KEY` is injected correctly and the flow can proceed. Does it make sense?

Reply:
In general => but I do not remember the code good enough to know when exactly are the secrets updated. The part it will likely fail are the rolling updates, bot the secret updates. So checking of the secrets works fine if they are updated only right before the roll.

Reply:
So on first roll, for trusting the new CA certificate, Secret is not update. On the second roll, the Secret is updated with the new CA certificate thumbprint right after rolling the pods.